### PR TITLE
Create main.go

### PR DESCRIPTION
**Pull request by faddat**
It's 75% yours anyway.  This is a simple STEEM -> rethinkdb JSON-eater.

### CODE REVIEW

File: examples/ingestron/main.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+func main() {
+	if err := run(); err != nil {
+		log.Fatalln("Error:", err)
+	}
+}
+
+// Set the settings for the DB
+func run() (err error) {
+    Rsession, err := r.Connect(r.ConnectOpts{
+    Addresses: []string{"192.168.194.98:28015","192.168.194.91:28015","192.168.194.145:28015","192.168.194.209:28015"},
```
Comment (tchap) [inline examples/ingestron/main.go:26]: @faddat Weird indentation, at least as I see it in the GitHub UI...

File: examples/ingestron/main.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    })
+    if err != nil {
+        log.Fatalln(err.Error())
```
Comment (tchap) [inline examples/ingestron/main.go:29]: @faddat It is enough to return the error, right?

File: examples/ingestron/main.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    }
+
+// Create a table in the DB
+var rethinkdbname string = "steemit"
```
Comment (tchap) [inline examples/ingestron/main.go:33]: @faddat I don't like the variable name. Should be `dbName` or something. Just use camelCase please...

File: examples/ingestron/main.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+_, err = r.DBCreate(rethinkdbname).RunWrite(Rsession)
+Rsession.Use(rethinkdbname)
+	if err != nil {
+		fmt.Println("rethindb DB already made")
```
Comment (tchap) [inline examples/ingestron/main.go:37]: @faddat Doesn't look like the most robust error check. Not sure how the DB driver works, but you are just assuming that getting an error here means that the DB already exists... IMO you should just fail on any error or check that the error is really that the DB already exists, if that is possible to check.

File: examples/ingestron/main.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+}
+	_, err = r.DB(rethinkdbname).TableCreate("blocks").RunWrite(Rsession)
+
+	if err != nil {
+		fmt.Println("Probably already made a table for blocks")
```
Comment (tchap) [inline examples/ingestron/main.go:42]: @faddat Same here...

File: examples/ingestron/main.go [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+	// Drop the error in case it is a request being interrupted.
+	defer func() {
+		if err == websocket.ErrClosing && interrupted {
+			err = nil
+		}
+	}()
+	// This allows you to tell the app which block to start on.
+	// TODO: Make all of the vars into a config file and package the binaries
+	Startblock := 3100000
```
Comment (tchap) [inline examples/ingestron/main.go:70]: @faddat Weird variable name, should be `startBlock`...

File: examples/ingestron/main.go [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+	U := uint32(Startblock)
```
Comment (tchap) [inline examples/ingestron/main.go:71]: @faddat Please use better variable names...

File: examples/ingestron/main.go [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+			}
+		}()
+	}
+
+	// Instantiate the WebSocket transport.
+	log.Printf("---> Dial(\"%v\")\n", url)
+	t, err := websocket.NewTransport(url,
+		websocket.SetAutoReconnectEnabled(reconnect),
+		websocket.SetAutoReconnectMaxDelay(30*time.Second),
+		websocket.SetMonitor(monitorChan))
```
Comment (tchap) [inline examples/ingestron/main.go:90]: @faddat I don't think that you need to use auto-reconnect and monitoring, it was just a demo in the example you based this code on. IMO it makes more sense to just drop this and also delete the extra code that is associated with this functionality. Examples should do as little as possible to be examples really...

File: examples/ingestron/main.go [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+	fmt.Println("---> Entering the block processing loop")
+	for {
+		// Get current properties.
+		props, err := client.Database.GetDynamicGlobalProperties()
+		if err != nil {
+			return err
+		}
+
+		// Process new blocks.
+		for props.LastIrreversibleBlockNum-U > 0 {
```
Comment (tchap) [inline examples/ingestron/main.go:134]: @faddat This code block is just horrible horrible. Please rewrite it to check for error codes and simply for the code not to look like a pile of rubbish.

File: examples/ingestron/main.go [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+			}
+
+			// Process the transactions.
+			if U != lastblock {
+				U++
+			}
+
+
+}
+time.Sleep(time.Duration(config.SteemitBlockInterval) * time.Second)
```
Comment (tchap) [inline examples/ingestron/main.go:155]: @faddat Weird indentation... Are you using `go fmt`?

File: examples/ingestron/main.go (hunk 8 above)
Comment (tchap) [reply examples/ingestron/main.go:90]: Well, if you want to keep inserting the block continuously, why not...

**Review by tchap — CHANGES_REQUESTED** [review]
Please address the issues before I can merge this...

**Comment by faddat** [comment]
Pretty sure that I did not intend to send this as a PR-- nevermind-- I did, now I remember.  This is a super-early ingestron that I did submit.  Thank you for the code reveiw.  I'll update and resubmit.  As for go fmt, yes..... but it looks funny to me as well.

File: examples/ingestron/main.go (hunk 4 above)
Comment (faddat) [reply examples/ingestron/main.go:37]: Hey, I just noticed this-- Thank you very much for the code review!

File: examples/ingestron/main.go (hunk 9 above)
Comment (faddat) [reply examples/ingestron/main.go:134]: lol-- my first go app :).  Thanks again :).

File: examples/ingestron/main.go (hunk 10 above)
Comment (faddat) [reply examples/ingestron/main.go:155]: just 3 or 4 weeks ago I got a good dev env running.  Shamefully, I was probably not using go fmt, or if I was, I wasn't using it right.  Again, many thanks.  I'm revising this app today, and will resubmit the PR in a much different form.

**Pull request closed without merging**